x/concentrated-liquidity/tick.go

```diff
@@ -14,7 +14,15 @@ import (
 	types "github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/types"
 )
 
+// initOrUpdateTick retrieves the tickInfo from the specified tickIndex and updates both the liquidityNet and LiquidityGross.
+// if we are initializing or updating an upper tick, we subtract the liquidityIn from the LiquidityNet
+// if we are initializing or updating an lower tick, we add the liquidityIn from the LiquidityNet
 func (k Keeper) initOrUpdateTick(ctx sdk.Context, poolId uint64, tickIndex int64, liquidityIn sdk.Dec, upper bool) (err error) {
+	_, err = k.getPoolById(ctx, poolId)
```

Comment:
Should we implement `hasPool` method?

Reply:
Do you mean just create a func in pool.go such as `func (k Keeper) hasPool(ctx sdk.Context, poolId uint64)` that just returns a bool if it exists or not?

Reply:
yeah

Reply:
okay, added a `poolExists` method!